include/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/Weather/CSSISpaceWeather.hpp

```diff
@@ -0,0 +1,170 @@
+        Real F107AdjLast81;    /// Last 81-day arithmetic average of F10.7 (adjusted)
+    };
+
+    // Note: daily prediction contains idential values to observations
+    using DailyPrediction = Observation;
+
+    struct MonthlyPrediction
+    {
+        Date date;     /// UTC day of reading.
+        Integer BSRN;  /// Bartels Solar Rotation Number. A sequence of 27-day intervals counted continuously from 1832
+                       /// Feb 8.
+        Integer ND;    /// Number of Day within the Bartels 27-day cycle (01-27).
+        Integer ISN;   /// International Sunspot Number. Records contain the Zurich number through 1980 Dec 31 and the
+                       /// International Brussels number thereafter.
+        Real F107Obs;  /// Observed 10.7-cm Solar Radio Flux (F10.7). Measured at Ottawa at 1700 UT daily from 1947 Feb
+                       /// 14 until 1991 May 31 and measured at Penticton at 2000 UT from 1991 Jun 01 on. Expressed in
+                       /// units of 10-22 W/m2/Hz.
+        Real F107Adj;  /// 10.7-cm Solar Radio Flux (F10.7) adjusted to 1 AU.
+        String F107DataType;   /// Flux Qualifier.
+                               /// OBS: Observed flux measurement
+                               /// INT: CelesTrak linear interpolation of missing data
+                               /// PRD: 45-Day predicted flux
+                               /// PRM: Monthly predicted flux
+        Real F107ObsCenter81;  /// Centered 81-day arithmetic average of F10.7 (observed).
+        Real F107ObsLast81;    /// Last 81-day arithmetic average of F10.7 (observed).
+        Real F107AdjCenter81;  /// Centered 81-day arithmetic average of F10.7 (adjusted).
+        Real F107AdjLast81;    /// Last 81-day arithmetic average of F10.7 (adjusted)
+    };
+
+    friend std::ostream& operator<<(std::ostream& anOutputStream, const CSSISpaceWeather& aCSSISpaceWeather);
+
+    bool isDefined() const;
+
+    const Date& accessLastObservationDate() const;
+
+    const Interval& accessObservationInterval() const;
+
+    const Observation& accessObservationAt(const Instant& anInstant) const;
+
+    const Interval& accessDailyPredictionInterval() const;
+
+    const DailyPrediction& accessDailyPredictionAt(const Instant& anInstant) const;
+
+    const Interval& accessMonthlyPredictionInterval() const;
+
+    const MonthlyPrediction& accessMonthlyPredictionAt(const Instant& anInstant) const;
+
+    static CSSISpaceWeather Undefined();
+
+    static CSSISpaceWeather Load(const File& aFile);
+
+   private:
+    Date lastObservationDate_;
+
+    Interval observationInterval_;
+    Map<Integer, CSSISpaceWeather::Observation> observations_;
+
+    Interval dailyPredictionInterval_;
+    Map<Integer, CSSISpaceWeather::DailyPrediction> dailyPredictions_;
+
+    Interval monthlyPredictionInterval_;
+    Map<Integer, CSSISpaceWeather::MonthlyPrediction> monthlyPredictions_;
+
+    CSSISpaceWeather();
```

Comment:
Sorry I didn't make this clear before but what is the point of having a private default constructor vs just the undefined static constructor?

Reply:
I see your point. This is a carry-over from the one I copied from, but I don't think we need it. Let me try it.

Reply:
Looks like prior to C++ 11, this was the preferred way to hide the default constructor. Probably a holdover from that. [ref](https://stackoverflow.com/questions/10474417/how-to-delete-the-default-constructor)

Reply:
Ah wait, I think you actually _do_ need both. 

If you remove the empty Constructor, there are no constructors left to call. the `CSSISpaceWeather::Undefined()` function has to call _something_, so this is needed.

You could alternatively get rid of the `Undefined()` function and just keep the default empty constructor, but I think that makes it a bit less clear that it is not defined.

Reply:
Sorry I meant that you remove the private constructor, and just have the Undefined static constructor. And you move the logic from the private constructor to the Static undefined constructor. This is because for most of OSTk we don't allow a default constructor. In this case it's private which is 🧷 but it's not used anywhere so it's kinda pointless (unless I'm missing something).

Reply:
Maybe I'm not understanding the pattern that you mean. From what I understand, `Undefined()` is not a constructor, but a static factory function that calls a constructor. 

In other cases that I've seen, we always have at least one non-default constructor, then `Class::Undefined()` can call that with undefined arguments.
In this case if we remove the default constructor, there are no non-default constructor so `Class::Undefined()` has nothing to call and it can't make an object.

Very possible this is a gap in my cpp-fu 🙏

Reply:
Ah yeah fair point that there is no other constructor, I missed that sorry. I'm going to push a suggestion commit on top to match a better pattern in my opnion.

Reply:
Sounds good, although I would say that if we're changing the pattern, we should probably do it in all of the Managers so that it remains a pattern, no? 

What kinda pattern you thinking? I'm happy to help with that!

Reply:
We hashed it out in this thread https://loftorbital.slack.com/archives/C03GCNLG8G5/p1687393460392469

Decided that the CSSISpaceWeather class might not be the best place to inject test data. Got offline :blessing: from Vishwa